src/main/java/org/jenkinsci/plugins/configfiles/buildwrapper/ManagedFileUtil.java

```diff
@@ -97,12 +105,12 @@ public FilePath call() throws IOException {
 
             FilePath target = null;
             if (createTempFile) {
-                target = ManagedFileUtil.createTempFile(build.getWorkspace().getChannel());
+                target = ManagedFileUtil.createTempFile(ws.getChannel());
             } else {
                 
                 String expandedTargetLocation = managedFile.targetLocation;
                 try {
-                    expandedTargetLocation = TokenMacro.expandAll(build, listener, managedFile.targetLocation);
+                    expandedTargetLocation = TokenMacro.expandAll((AbstractBuild<?, ?>) build, listener, managedFile.targetLocation);
```

Comment:
Of course this will produce a `ClassCastException` if called from a `WorkflowRun`.

https://github.com/jenkinsci/email-ext-plugin/pull/97 seemed to require a similar API change in `token-macro-plugin` which @tfennelly was working on at some point.